dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+# Demystifying `aui:script` and `liferay-frontend:component` tags
+
+On JSP context you may notice a huge number of `aui:script` and `liferay-frontend:component` tags.
+
+## `aui:script` tag
+
+### What it does:
+
+This taglib allow users to use and require another JS module/resources inside a JSP.
+
+This taglib have three possible use cases:
+
+1. Pure JS on the JSP
+2. Import a AUI/YUI module
+3. Import a ES6 module
+
+#### Pure JS on the JSP
+
+By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle, see [combo for more details](https: / /github.com/liferay/liferay-frontend-guidelines/blob/45bad0334cb467e45e68aee322a2a4bc0b6979d3/dxp/javaScript_minification.md#comboservlet). It is worth mentioning that the same rule applies for the following use cases.
```

Comment:
Unless I am misunderstanding how this works, I don't think the combo servlet has much to do with this at all. `aui:script` tags don't have their JS extracted out in order to get served separately via the combo servlet; they get spat out into the HTML of the page markup itself.

Also, that link looks broken (spaces around the `/`, and the case of the `S` looks wrong).

Reply:
> Again here I'd drop "Pure" and say "Inline" (or "embedded")?

> ("Pure" is kind of an implementation detail, as it hints at whether or not the JS is transpiled or mixed up with other stuff; in our case it is not transpiled but it's not "pure" either, because it can have JSP expressions, comments, declarations, directives, scriptlets etc interpolated inside it.)

I adopted "Inline" instead of "Pure"